docs/types.rst

```diff
@@ -500,12 +502,14 @@ Another example that uses external function types::
 
 .. index:: ! type;reference, ! reference type, storage, memory, location, array, struct
 
+.. _reference-types:
+
 Reference Types
 ==================
 
 Complex types, i.e. types which do not always fit into 256 bits have to be handled
-more carefully than the value-types we have already seen. Since copying
-them can be quite expensive, we have to think about whether we want them to be
+more carefully than the value-types we have already seen. `Since copying them <ref-copy-types_>`_
```

Comment:
I think the link should only cover `copying`, shouldn't it?

Reply:
Should “Complex Types” be “Dynamic Types” for consistency?

Reply:
I think it should be either `complex types` or `reference types`.

Reply:
@maurelian Hmm, my reasoning here (and it looks like this was already the approach of others) was that a 'complex type' (with the definition as above) is referenced because it is 'complex'. So they are the same but use different terms for clearer explanation.

I don't know if that clarifies at all 😆 .

For your comment @chriseth are you saying that you're happy with the terms used, or that we should just stick to one term and explain why with different terms?

Reply:
We should stick to reference types. All complex types are reference types, but that might change in the future. A complex type is a type that is composed out of other types.

Basically the issue here is that the section is called "reference types" but for whatever reason we are talking about complex types and that is confusing, I would say.